pkg/api/deploymentapi/deploymentsize/sizes.go

```diff
@@ -0,0 +1,65 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymentsize
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const minsize = 512
+
+// Parse converts the stringified size notation to an int32 Megabyte value.
+// The minimum size allowed is 512 Megabytes.
+func Parse(strSize string) (int32, error) {
```

Comment:
As far as I know the only allowed values are 512 and multiples of 1024. We should account for that when the input is in MB. Would be good to return an error when the size is anything but the values mentioned above.

Reply:
I didn't want to be writing rules which might change in the future, and let the API do the validation when the API call is performed. Not sure if I'm being too conservative